src/tools/clippy/clippy_lints/src/unnested_or_patterns.rs

```diff
@@ -392,26 +402,42 @@ fn extend_with_tail_or(target: &mut Pat, tail_or: Vec<P<Pat>>) -> bool {
 // Only elements beginning with `start` are considered for extraction.
 fn drain_matching(
     start: usize,
-    alternatives: &mut Vec<P<Pat>>,
+    alternatives: &mut ThinVec<P<Pat>>,
     predicate: impl Fn(&PatKind) -> bool,
     extract: impl Fn(PatKind) -> P<Pat>,
-) -> Vec<P<Pat>> {
-    let mut tail_or = vec![];
+) -> ThinVec<P<Pat>> {
+    let mut tail_or = ThinVec::new();
     let mut idx = 0;
-    for pat in alternatives.drain_filter(|p| {
-        // Check if we should extract, but only if `idx >= start`.
+
+    // If `ThinVec` had the `drain_filter` method, this loop could be rewritten
```

Comment:
Is changing things in Clippy to `ThinVec` necessary? Just looking at the Clippy changes seems that this just makes things worse, from a code style perspective. I would like if this could be avoided in Clippy and keep the old cleaner code.

Reply:
It can't be avoided. Clippy uses rustc's AST types. When those AST types change, Clippy has to change too.